@patch('tarfile.open')
@patch('os.path.isdir')
@patch('os.path.exists')
@patch('os.path.isfile')
@patch('os.remove')
@patch('os.chmod')
@patch('shutil.rmtree', new=MagicMock())
@patch('glob.iglob', new=MagicMock(return_value=['/usr/hdp/2.2.1.0-2187/hadoop/lib/hadoop-lzo-0.6.0.2.2.1.0-2187.jar']))
@patch('shutil.copy')
@patch.object(shell, 'call')
def test_upgrade_failed_prepare_war(self, call_mock, shutil_copy_mock, chmod_mock, remove_mock, isfile_mock, exists_mock, isdir_mock, tarfile_open_mock):
    isdir_mock.return_value = True
    exists_mock.side_effect = [False, False, True]
    isfile_mock.return_value = True
    call_mock.return_value = (0, 'Whoops, you messed up the WAR.')
    try:
        self.executeScript((self.COMMON_SERVICES_PACKAGE_DIR + '/scripts/oozie_server.py'), classname='OozieServer', command='pre_rolling_restart', config_file='oozie-upgrade.json', hdp_stack_version=self.UPGRADE_STACK_VERSION, target=RMFTestCase.TARGET_COMMON_SERVICES)
        self.fail('An invalid WAR preparation should have caused an error')
    except Fail as f:
        pass
